Proteomics/Modification.cs

```diff
@@ -36,12 +36,14 @@ public override string ToString()
         public override bool Equals(object o)
         {
             Modification m = o as Modification;
-            return o != null;
+            return o != null 
+                && m.id == this.id 
+                && m.modificationType == this.modificationType;
         }
 
         public override int GetHashCode()
         {
-            return 0;
+            return id.GetHashCode();
```

Comment:
Note, this won't work if we allow null IDs. Should we disallow null for ID in the constructor?